python/pyarrow/_compute.pyx

```diff
@@ -785,6 +785,30 @@ class ElementWiseAggregateOptions(_ElementWiseAggregateOptions):
         self._set_options(skip_nulls)
 
 
+cdef CInclusive unwrap_inclusive(inclusive) except *:
+    if inclusive == "both":
+        return CInclusive_BOTH
+    elif inclusive == "left":
+        return CInclusive_LEFT
+    elif inclusive == "right":
+        return CInclusive_RIGHT
+    elif inclusive == "neither":
+        return CInclusive_NEITHER
+    _raise_invalid_function_option(inclusive, "inclusive")
+
+
+cdef class _BetweenOptions(FunctionOptions):
+    def _set_options(self, inclusive):
+        self.wrapped.reset(
+            new CBetweenOptions(unwrap_inclusive(inclusive))
+        )
+
+
+class BetweenOptions(_BetweenOptions):
```

Comment:
Can you add a docstring to this class? (it's a recent change in master that all classes now have a docstring, see the other classes in this file)